scipy/optimize/tests/test_zeros.py

```diff
@@ -58,5 +60,18 @@ def test_deriv_zero_warning(self):
         assert_warns(RuntimeWarning, cc.newton, func, 0.0, dfunc)
 
 
+class TestRootResults:
+    def test_repr(self):
+        _, r = zeros.bisect(functions[0], .5, sqrt(3),
+                            xtol=0.1e-12, full_output=True)
+        expected_repr = ("      converged: True\n           flag: 'converged'"
+                         "\n function_calls: 46\n     iterations: 44\n"
+                         "           root: 0.999999999999985")
+        # Pre-2.7, float repr had a fixed 17 digits of precision
+        if version_info < (2, 7):
+            expected_repr = expected_repr + '01'
```

Comment:
This will not necessary guarantee it works --- the rounding error can depend on the platform, so people may still see the message if they run the tests on their own system. A better way around it: use 
```
r.root = 1.0
r.iterations = 44
r.function_calls = 46
```
to set all values that potentially depend on rounding to known numbers.